commercetools-test-lib/src/main/java/io/sphere/sdk/test/ApacheHttpClientAdapterImpl.java

```diff
@@ -0,0 +1,133 @@
+package io.sphere.sdk.test;
+
+import io.sphere.sdk.http.*;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.input.AutoCloseInputStream;
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.FileEntity;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
+import org.apache.http.message.BasicNameValuePair;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPInputStream;
+
+import static java.util.Arrays.asList;
+
+final class ApacheHttpClientAdapterImpl extends HttpClientAdapterBase {
```

Comment:
How about renaming this class to `IntegrationTestHttpClient`? And adding javadoc with a short explanation that we had to copy this class for our integration tests to avoid cyclic dependencies? What do you think?